File: python/src/wis/auth/logout.py
import requests
# #############################################
from ..wis_constants import wis_base_url
from ..wis_exception import WISException
from .token import Token
# #############################################


def logout(token: Token):
    """ Logout

        Inputs:
            token (Token):      ...
    """

    url = wis_base_url + "/logout"
    payload = {}
    headers = { **token.header }
    response = requests.request("POST", url, headers=headers, data=payload)

    if response.status_code != 204:
        raise WISException.from_response(response)
# #############################################

File: typescript/samples/auth/logout.ts
import axios, { AxiosRequestConfig, AxiosResponse } from "axios";
import { WisError } from "../../types";

export async function logout<T>(token: string): Promise<T | null> {
  const config: AxiosRequestConfig<any> = {
    method: "post",
    url: "https://api.services.wassa.io/logout",
    headers: {
      Authorization: "Bearer " + token,
    },
  };

  try {
    const response: AxiosResponse<any> = await axios(config);
    return response.data;
  } catch (error: any) {
    throw error as WisError;
  }
}
